extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/ConflictsWithScrutinizer.java

```diff
@@ -36,25 +66,29 @@ public void scrutinize(ItemUpdate update) {
         }
 
         for(PropertyIdValue propertyId : propertyIdValueValueMap.keySet()){
-            Map<PropertyIdValue, List<Value>> conflictingPropertyMap = _fetcher.getParamConflictsWith(propertyId);
-            for (PropertyIdValue conflictingPid : conflictingPropertyMap.keySet()) {
-                if (propertyIdValueValueMap.containsKey(conflictingPid) && raiseWarning(propertyIdValueValueMap, conflictingPid, conflictingPropertyMap)) {
+            List<Statement> statementList = _fetcher.getConstraintsByType(propertyId, CONFLICTS_WITH_CONSTRAINT_QID).collect(Collectors.toList());
+            for (Statement statement : statementList) {
+                ConflictsWithConstraint constraint = new ConflictsWithConstraint(statement);
```

Comment:
What happens when the statement does not contain the necessary qualifiers to extract constraint parameters? For instance, a conflicts with constraint without any parameter? We should make sure such declarations are ignored. This will be easier to test with once we migrate to proper mocking, because you will be able to control more easily the return value of `getConstraintsByType` in each test.

Reply:
For now, In that case conflictingPid will be null and `propertyIdValueValueMap.containsKey(conflictingPid)` this will return false and the if condition will not execute in that case. Is it fine for now or it needs some changes?

Reply:
Let's just try to remember to add a test for it once we have proper mocking in place :)

Reply:
Sure!